OPHD/States/MapViewState.cpp

```diff
@@ -228,17 +228,13 @@ void MapViewState::initialize()
 
 	mPopulationPool.population(&mPopulation);
 
+	StructureCatalogue::init(mPlanetAttributes.meanSolarDistance);
+	ProductCatalogue::init("factory_products.xml");
+
 	if (mLoadingExisting)
 	{
 		load(mExistingToLoad);
 	}
-	else
-	{
-		// StructureCatalogue is initialized in load routine if saved game present to load existing structures
-		StructureCatalogue::init(mPlanetAttributes.meanSolarDistance);
```

Comment:
I believe `mPlanetAttributes` isn't guaranteed to be initialized until after the call to `load`. There are two constructors to `MapViewState`. One of them initializes `mPlanetAttributes`, while the other sets `mLoadingExisting` and defers initialization to the `load` call.

It's weird, and probably should be changed, but if that hasn't been accounted for, this may be introducing a bug.